cmake/modules/FindMaya.cmake

```diff
@@ -69,21 +68,14 @@ else(IS_LINUX)
 endif()
 
 if(IS_MACOSX)
-    # Note: according to official Autodesk sources (and how it sets up
-    # MAYA_LOCATION itself), MAYA_LOCATION should include Maya.app/Contents
-    # on MacOS - ie:
-    #   /Applications/Autodesk/maya2019/Maya.app/Contents
-    # However, for legacy reasons, and for maximum compatibility, setting
-    # it to the installation root is also supported, ie:
-    #   /Applications/Autodesk/maya2019
-
```

Comment:
I admit having two executable is really confusing on MacOSX and that is something that Autodesk needs to fix in future release.

There are two executable for Maya on MacOSx.  `Contents/MacOS/Maya` and ` Contents/bin/maya`.

`bin` directory is not really a normal macOS bundle item.  `Contents/MacOS` is the actual bundle binary that we build, all of the internal rpath are relative to and is what we encourage users to use.  Currently this FindMaya module is setting `MAYA_EXECUTABLE ` based on `Maya.app/Contents/bin`. I didn't think it is necessary to change this for the sake of this PR but that's something that we need to do later.

Reply:
We don't officially recommend "MAYA_LOCATION should include Maya.app/Contents". Can you point to me to the official page that mentions this? In a MacOS application bundle, the first directory in the bundle underneath the top-level directory is "Contents".

> it might be helpful to keep a note saying where MAYA_LOCATION should be set to

I agree. On MacOSX, I normally just have `--maya-location` pointing to where Maya.app sits `e.g /Application/Autodesk/maya20xx`. The MAYA_EXECUTABLE then is picked up based on PATH_SUFFIXES 

```
find_program(MAYA_EXECUTABLE
        maya
    HINTS
        "${MAYA_LOCATION}"
        "$ENV{MAYA_LOCATION}"
        "${MAYA_BASE_DIR}"
    PATH_SUFFIXES
        Maya.app/Contents/bin/
        bin/
    DOC
        "Maya's executable path"
)
```

We even have this super confusing official recommendation like this which we still carry over to this day:
```
On Mac OS X, Maya is installed as /Applications/Maya <version>/Maya.app. Since Mac OS X architecture makes it difficult to access the contents of the Maya application package, we strongly discourage the use of MAYA_LOCATION on Mac OS X. Instead, use the following alternate locations: 
```
https://knowledge.autodesk.com/support/maya/getting-started/caas/CloudHelp/cloudhelp/2019/ENU/Maya-EnvVar/files/GUID-228CCA33-4AFE-4380-8C3D-18D23F7EAC72-htm.html

Reply:
Well, the most "official" source I could think of is what maya itself does - if you launch a default-install Maya.app (and you don't have MAYA_LOCATION set yourself in your env, or anything), you'll see that it sets it to `Maya.app/Contents`:

```
import os
print os.environ['MAYA_LOCATION']
/Applications/Autodesk/maya2019/Maya.app/Contents
```

(Note - as you can see, I tested with 2019, as I haven't installed 2020 on my mac laptop yet.  But I'd doubt if it's changed...)

It also used to say this in the "maya" launch script, as well - back when it was just a text bash script file.  (It's binary now, so I can't read it.) There is verbiage to this effect in the devkit, as well - in devkitBase/devkit/bin/mayald, there is this line:

```
# NOTE: MAYA_LOCATION on the Mac OS X includes the path to the
# Maya.app/Contents directory.
```

However - I did notice when searching the devkit that the devkit/README.md states:

> create the environment variable `MAYA_LOCATION` and set it to where Maya is installed (default location:  `/usr/autodesk/maya2020`)

Also, the FindMaya.cmake (and other cmake files there) do seem to use the "base install dir" definition.  So it seems that as far as official Autodesk cmake files, go, I stand corrected.  I guess I just assumed that the definition for cmake's MAYA_LOCATION would be the same as the environ's MAYA_LOCATION, since it was a pretty long standing convention.

Reply:
@elrond79 
There is certainly some vagueness here.  There are so many important directories at the application level: e.g `include`, `plug-ins`, etc.... 

So for example, let's say for whatever reasons I want to refer to Maya's `plug-ins` directory and since MAYA_LOCATION is set to `Maya.app/Contents` then I have do `../../plug-ins`. That's seems just weird!

I mean, we are not trying to solve Goldbach’s conjecture﻿ here but MacOSX directory structure needs a bit of organization and consistency between the docs would be nice. I have already discussed this internally with the foundation team and there are agreements on changing this.

Reply:
Ok... well, maybe we can just add a quick note like this?

```
# On OSX, setting MAYA_LOCATION to either the base installation dir (ie,
# `/Application/Autodesk/maya20xx`), or the Contents folder in the Maya.app dir
# (ie, `/Application/Autodesk/maya20xx/Maya.app/Contents`) are supported.
```

Otherwise, it's a bit unclear why we're checking `../..`.

Also, if we want to make the installation root be the new standard for MAYA_LOCATION on MacOS, we might want to flip the order that it tries the paths, so it tries `${MAYA_LOCATION}` first, then `${MAYA_LOCATION}/../..`

Reply:
@elrond79  Done! Thank you, the wording makes sense to me. See 96156e7ab